torch2jax/__init__.py

```diff
@@ -242,15 +256,39 @@ def fn(*args, **kwargs):
 auto_implements(torch.add, jnp.add)
 auto_implements(torch.exp, jnp.exp)
 auto_implements(torch.nn.functional.gelu, jax.nn.gelu)
-auto_implements(torch.mean, jnp.mean)
 auto_implements(torch.mul, jnp.multiply)
 auto_implements(torch.permute, jnp.transpose, dont_coerce_argnums=(1, 2))  # TODO: do we need argnum 2?
 auto_implements(torch.pow, jnp.power)
+auto_implements(torch.sin, jnp.sin)
+auto_implements(torch.cos, jnp.cos)
 auto_implements(torch.sigmoid, jax.nn.sigmoid)
 auto_implements(torch.sqrt, jnp.sqrt)
-auto_implements(torch.sum, jnp.sum)
 auto_implements(torch.tanh, jnp.tanh)
 auto_implements(torch.transpose, jnp.swapaxes)
+auto_implements(torch.rsqrt, lambda x: jnp.reciprocal(jnp.sqrt(x)))
+
+
+@implements(torch.nn.functional.silu)
+def silu(x, *args, **kwargs):
+  x = _v(x)
+  if "inplace" in kwargs and kwargs["inplace"]:
+    assert not kwargs["inplace"], "jax doesn't support inplace"
+  return jax.nn.silu(x)
+
+
+@implements(torch.device, Torchishify_output=False)
+def torch_device(type):
```

Comment:
```suggestion
def device(type):
```

Reply:
what would a test case that requires this look like? doesn't this just auto-fallback to the normal torch.device implementation?

Reply:
Yes, this autofalls back, due to the above `torch.device("cpu")` call would be intercepted by `__torch_function__` mechanism, that's why the need for this function. I think we just need to keep the original `torch.device` and can not replace with `jax.Device` here because the code that consume this object are python code that are not part of `__torch_function__`.